src/fondant/component/component.py

```diff
@@ -73,3 +75,30 @@ def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
 
 Component = t.TypeVar("Component", bound=BaseComponent)
 """Component type which can represents any of the subclasses of BaseComponent"""
+
+
+def python_component(
+    extra_requires: t.Optional[t.List[str]] = None,
+    base_image: t.Optional[str] = None,
+):
+    """Decorator to enable a python component."""
+
+    def wrapper(cls):
+        image = Image(
+            base_image=base_image,
+            extra_requires=extra_requires,
+        )
+
+        @wraps(cls, updated=())
+        class PythonComponent(cls):
+            @classmethod
+            def image(cls) -> Image:
+                return image
+
+            @classmethod
+            def is_python_component(cls) -> bool:
+                return True
+
+        return PythonComponent
+
+    return wrapper
```

Comment:
It works but I'm not completely convinced this is the way forward. Essentially we subclass the Component and add an image property with the details provided in the decorator arguments.